lightstep/util.go

```diff
@@ -21,3 +25,21 @@ func convertNestedMapToSchemaSet(opts map[string]interface{}) *schema.Set {
 	}
 	return schema.NewSet(f, []interface{}{opts})
 }
+
+// Note due to Terraform's issues with TypeMap having TypeBool elements, we
+// need to use boolean strings
+func setHiddenQueriesFromResourceData(qs map[string]interface{}, query client.MetricQueryWithAttributes) {
+	if len(query.HiddenQueries) == 0 {
+		// nothing to do
+		return
+	}
+	hq := make(map[string]interface{}, len(query.HiddenQueries))
+	for k, v := range query.HiddenQueries {
+		// Don't include the top-level query in the TF resource data
```

Comment:
note that this leads to another unresolvable diff scenario, specifically when the user specifies a `hidden_queries` with the top-level query name included. i am choosing not to fix this in the provider, because ~i dont know how~ i think the API should be updated to better handle this case -- in particular, we should omit the top-level query name from `hidden_queries` in the API, and use `hidden` for that purpose (as the contract has been defined here)